test/test_codec.py

```diff
@@ -24,175 +27,175 @@ def test_bitstruct_format(self):
 
-        s4 = Signal('signal', signalSize=8, is_little_endian=False)
+        s4 = Signal('signal', size=8, is_little_endian=False)
         self.assertEqual(s4.bitstruct_format(), '>s8')
 
     def test_encode_signal(self):
-        s1 = Signal('signal', signalSize=8)
+        s1 = Signal('signal', size=8)
         self.assertEqual(s1.phys2raw(), 0)
         self.assertEqual(s1.phys2raw(1), 1)
         self.assertEqual(s1.phys2raw(s1.max), 127)
         self.assertEqual(s1.phys2raw(s1.min), -128)
 
-        s2 = Signal('signal', signalSize=10, is_signed=False)
+        s2 = Signal('signal', size=10, is_signed=False)
         self.assertEqual(s2.phys2raw(), 0)
         self.assertEqual(s2.phys2raw(10), 10)
         self.assertEqual(s2.phys2raw(s2.max), 1023)
         self.assertEqual(s2.phys2raw(s2.min), 0)
 
-        s3 = Signal('signal', signalSize=8, factor=2)
+        s3 = Signal('signal', size=8, factor=2)
         self.assertEqual(s3.phys2raw(), 0)
         self.assertEqual(s3.phys2raw(10), 5)
         self.assertEqual(s3.phys2raw(s3.max), 127)
         self.assertEqual(s3.phys2raw(s3.min), -128)
 
-        s4 = Signal('signal', signalSize=8, is_signed=False, factor=5)
+        s4 = Signal('signal', size=8, is_signed=False, factor=5)
         self.assertEqual(s4.phys2raw(), 0)
         self.assertEqual(s4.phys2raw(10), 2)
         self.assertEqual(s4.phys2raw(s4.max), 255)
         self.assertEqual(s4.phys2raw(s4.min), 0)
 
-        s5 = Signal('signal', signalSize=8, offset=2)
+        s5 = Signal('signal', size=8, offset=2)
         self.assertEqual(s5.phys2raw(), 0)
         self.assertEqual(s5.phys2raw(10), 8)
         self.assertEqual(s5.phys2raw(s5.max), 127)
         self.assertEqual(s5.phys2raw(s5.min), -128)
 
-        s6 = Signal('signal', signalSize=8, is_signed=False, offset=5)
+        s6 = Signal('signal', size=8, is_signed=False, offset=5)
         self.assertEqual(s6.phys2raw(), 0)
         self.assertEqual(s6.phys2raw(10), 5)
         self.assertEqual(s6.phys2raw(s6.max), 255)
         self.assertEqual(s6.phys2raw(s6.min), 0)
 
-        s7 = Signal('signal', signalSize=8)
+        s7 = Signal('signal', size=8)
         s7.addAttribute('GenSigStartValue', '5')
         self.assertEqual(s7.phys2raw(), 5)
 
-        s8 = Signal('signal', signalSize=8, is_signed=False, offset=5)
+        s8 = Signal('signal', size=8, is_signed=False, offset=5)
         s8.addAttribute('GenSigStartValue', '5')
         self.assertEqual(s8.phys2raw(), 5)
 
-        s9 = Signal('signal', signalSize=16, is_signed=False, factor=0.001)
+        s9 = Signal('signal', size=16, is_signed=False, factor=0.001)
         self.assertEqual(s9.phys2raw(), 0)
-        self.assertEqual(s9.phys2raw(s9.max), 65534)
+        self.assertEqual(s9.phys2raw(s9.max), 65535)
         self.assertEqual(s9.phys2raw(s9.min), 0)
-        self.assertEqual(s9.phys2raw(50.123), 50123)
+        self.assertEqual(s9.phys2raw(D(50123)*D(0.001)), 50123)
```

Comment:
As noted before, pass a string to `decimal.Decimal` instead of a float to avoid loss of accuracy.